### ISSUE TITLE
No hostname(s) and/or IP address(es) is registered on backend's URL when the Docker service is created as --endpoint-mode dnsrr in Docker swarm mode

### ISSUE THREAD

**Issue by bintut** [containous/traefik#1494]
### Do you want to request a *feature* or report a *bug*?

I think this is a bug.

### What did you do?

I created the Docker service in `--endpoint-mode dnsrr` with 3 replicas as the backend service. The frontend's rule is `PathPrefix`. If I remove the `--endpoint-mode dnsrr` which means that the endpoint mode will be in VIP, it will automatically register the VIP in the backend's URL.

### What did you expect to see?

The service name or the IP addresses of all the tasks (containers) created should be registered on the backend's URL.

### What did you see instead?

There is no address in the URL. It just show as http://:$port_number

### Output of `traefik version`: (_What version of Traefik are you using?_)

[Traefik Linux AMD64 v1.2.3](https://github.com/containous/traefik/releases/download/v1.2.3/traefik_linux-amd64)

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "example.com"
watch = true
exposedbydefault = true
usebindportip = true
swarmmode = true
```

**Comment by ldez** [comment]
Seems to be a duplicate of #1463

**Comment by bintut** [comment]
I don't think this issue is a duplicate to #1463 for 3 main reasons:

1) Traefik is running as a Docker service with at least 3 replicas and using the default --endpoint-mode as VIP on the same swarm cluster and overlay network;
2) My backend services are running in --endpoint-mode dnsrr which does not publish ports outside of its cluster;
3) There is no hostname or IP address in my backend's URL, meaning only the transport protocol and port number appears in the URL (e.g. http://:12345).

**Comment by samar1n** [comment]
Same problem here.
`level=debug msg="Creating server server-r-backend-1-c5d61adc at http://:80 with weight 0" `

**Comment by bintut** [comment]
If the `--endpoint-mode` is **VIP** (which is the default) in Docker swarm mode (backend), the Traefik labels `traefik.backend.loadbalancer.sticky` and `traefik.backend.loadbalancer.method` are useless. That is why in order to make use of these labels and their features, we need to use the `--endpoint-mode` as **dnsrr**. Unfortunately, this bug is somehow a blocker for us.

**Comment by jamroks** [comment]
I'm facing the same issue here, as described by @bintut  
here is my compose yml file in v3.4 format:
``
whoami-ingress:
    image: emilevauge/whoami
    deploy:
      endpoint_mode: dnsrr
      mode: replicated
      replicas: 2
      placement:
          constraints: [ node.role == worker ]
      labels:
             - "traefik.enable=true"
             - "traefik.logLevel=DEBUG"
             - "traefik.frontend.rule=Host:whoami.labz.roks"
             - "traefik.port=80"
             - "traefik.backend=whoami"
             - "traefik.passHostHeader=true"
             - "traefik.frontend.entryPoints=http"
             - "traefik.backend.loadbalancer.swarm=true"
             - "traefik.backend.loadbalancer.method=drr"
    networks:
      - traefik_net
``

**Comment by jamroks** [comment]
Here is the content of my traefik.toml and docker version is : 
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:41:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:49 2017
 OS/Arch:      linux/amd64
 Experimental: false

```toml
# traefik.toml
################################################################
# Global configuration
################################################################

logLevel = "DEBUG"
defaultEntryPoints = ["http"]
[entryPoints]
  [entryPoints.http]
  address = ":80"

[web]
address = ":8080"
[web.auth.basic]
  users = ["admin:$apr1$8WpDE4Vj$kg80BCH1uB9kiZrS8tlyZ0"]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "labz.roks"
watch = true
swarmmode = true
exposedbydefault = false
```

**Comment by jamroks** [comment]
And here is the traefik UI showing that the backend lack an ip adress (http//:80) , so I guess that's the problem here. since it can't resolve the service IPs address, the response after a few second end up being : Bad Gateway !

<img width="1157" alt="screen shot 2017-10-29 at 17 14 26" src="https://user-images.githubusercontent.com/357258/32145682-108f8260-bccd-11e7-9f44-f51be4b83571.png">

Same result with a curl call : 

``` 
curl -v  -H Host:whoami.labz.roks http://172.18.30.30

 About to connect() to 172.18.30.30 port 80 (#0)
   Trying 172.18.30.30...
 Connected to 172.18.30.30 (172.18.30.30) port 80 (#0)
 GET / HTTP/1.1
 User-Agent: curl/7.29.0
 Accept: */*
 Host:whoami.labz.roks

 HTTP/1.1 502 Bad Gateway
 Content-Length: 11
 Content-Type: text/plain; charset=utf-8
 Date: Tue, 24 Oct 2017 21:52:35 GMT
 Connection #0 to host 172.18.30.30 left intact
```

<img width="127" alt="screen shot 2017-10-29 at 17 21 44" src="https://user-images.githubusercontent.com/357258/32145738-b4f0a938-bccd-11e7-9ade-c9b05d9f74f9.png">

**Comment by traefiker** [comment]
Closed by #2490.
